### ISSUE TITLE
Svelte 4.x support

### ISSUE THREAD

**Issue by Sinderella** [bryanmylee/svelte-headless-table#138]
Hi, just a heads-up, Svelte has released version 4. I don't think there's any incompatibility as I could run my app just fine, but there's a warning when upgrading my dependencies.

I see @bryanmylee is the author of both packages so I just raised it here.

```
 WARN  Issues with peer dependencies found
.
└─┬ svelte-headless-table 0.17.4
  └─┬ svelte-keyed 1.1.6
    └── ✕ unmet peer svelte@^3.49.0: found 4.1.2
```

Cheers!

**Comment by bryanmylee** [comment]
Hey, thanks for the report. I'll update `svelte-keyed` and this package.

**Comment by bryanmylee** [comment]
Closed by #141
